practice-areas/engineering/security-compliance.md

```diff
@@ -55,17 +55,92 @@ There are some instructions at <https://support.google.com/accounts/answer/18583
+
+Developers and themers working on the site server instance (SSH/shell, file system, database) must also:
+
+-   Follow best practices with respect to SSH keys, passphrases and passphrase caching (see above).
+-   Access the server only by methods (e.g. SSH, SFTP, SCP) configured by designated admins. Access by password, manually installed SSH keys (other than by admins), web based "shell" script, port forwarding to 3rd parties or other methods are forbidden, unless authorized in advance by the CTO.
+-   Restrict SSH port forwarding to temporary use for the purpose of accessing the server MySQL from your own desktop.
+-   Prefer the initiation of SSH connections from CivicActions servers to 3rd party servers, avoiding the reverse as much as possible.
+-   Obtain prior approval from a member of the IT team before running non-standard software on a server instance. This includes:
+    -   Daemons (persistent, long running processes)
+    -   Binary software (compiled on the server instance or elsewhere)
+    -   Web accessible scripts/CGIs that do not use solely an established framework
+-   Inform the IT team as soon as possible if unusual resource usage is anticipated, so that we can monitor resource utilization and ensure backup processes run correctly. This can include high traffic events, large data/media file uploads or high CPU/RAM usage (e.g. during large imports).
+
+IT team system administrators working on CivicActions servers must also:
+
+-   Take the utmost caution when working on server configuration - document and test each change.
+-   Non-urgent yet risky changes (those with significant risk of introducing undesired side-effects) should only be made when the person expects to remain online and available for a while after the change.
+-   Not work on site/user files as root - but "su" to the account first.
+-   Respect the privacy of server users, avoiding accessing others' personal data such as e-mails.
+-   Work with the IT team to ensure server and backup health is monitored and alerts are responded to promptly.
+-   Ensure offsite backups are transferred and stored only in encrypted form.
+-   Ensure the Hurricane Electric and RimuHosting access list (that controls remote hands and physical server access) is maintained.
+
 ## Continuous Monitoring
 
 We use tools to support continuous monitoring for performance and efficiency, and to ensure proper operation and security. These tools include (not an exhaustive list):
 
 -   Event and error log capture: auditd (SELinux), fail2ban and AIDE.
 -   Continuous monitoring dashboards: LogWatch, LogStash/GrayLog, Nagios, Cloudwatch, Pingdom, StatusCake, OpsGenie and New Relic.
--   Automated security scanning: [OpenSCAP](http://www.open-scap.org/), [ansible-role-800-53](https://github.com/RedHatGov/ansible-role-800-53), and [GovReady](https://github.com/GovReady).
+-   Automated security scanning: OpenSCAP, OWASP ZAP, Trivy and Syft (for SBOM creation).
```

Comment:
Recommend adding Dependency Track for the SBOM portion of the list. Maybe consider a separate line for SBOM generation and tracking as it's kind of, but not really security scanning?

Reply:
Added: "Supply chain and Software Bill of Materials (SBOM): Syft and DependencyTrack."